File: SistemaPedidosReact.Server/Models/OrderDetail.cs
using System.ComponentModel.DataAnnotations;

namespace SistemaPedidosReact.Server.Models
{
    public class OrderDetail
    {
        [Key]
        public int Id { get; set; }
        public int OrderId { get; set; }
        public string? DeliveryOperationType { get; set; } = string.Empty; //Turbo || Regular
        public DateTime CreatedAt { get; set; }
        public string? DeliveryMethod { get; set; } = string.Empty; //Mesa || Delivery || Pickup
        public int? MesaId { get; set; }
        public string? CantidadCubiertos { get; set; } = null;
        public int? PaymentMethodId { get; set; } = null;
        public decimal? Tip { get; set; } = 0;
        public int? DeliveryInformationId { get; set; } = null;
        public int? BillingInformationId { get; set; } = null;
        public int? DeliveryDiscountId { get; set; } = null;
        public int? TotalsId { get; set; } = null;

        public Order Order { get; set; } = null!;
        public Mesa? Mesa { get; set; } = null!;
        public PaymentMethod? PaymentMethod { get; set; } = null!;
        public DeliveryInformation? DeliveryInformation { get; set; }
        public BillingInformation? BillingInformation { get; set; }
        public DeliveryDiscount? DeliveryDiscount { get; set; } = null!;
        public Totals? Totals { get; set; } = null!;
        public virtual ICollection<OrderItem> OrderItems { get; set; } = new List<OrderItem>();
        public virtual ICollection<Discount> Discounts { get; set; } = new List<Discount>(); 
    }
}

File: sistemapedidosreact.client/src/interfaces/order-detail.ts
import type { BillingInformation } from "./billing-information";
import type { DeliveryDiscount } from "./delivery-discount";
import type { DeliveryInformation } from "./delivery-information";
import type { Discount } from "./discount";
import type { OrderItem } from "./order-item";
import type { Totals } from "./totals";

export interface OrderDetail {
  id: number;
  orderId: number;
  deliveryOperationType: string;
  createdAt: string; // ISO date string
  deliveryMethod: string;
  mesaId: number;
  cantidadCubiertos: string;
  paymentMethodId: number;
  tip: number;
  deliveryInformationId: number;
  billingInformationId: number;
  deliveryDiscountId: number;
  totalsId: number;
  deliveryInformation?: DeliveryInformation;
  billingInformation?: BillingInformation;
  deliveryDiscount?: DeliveryDiscount;
  totals: Totals;
  orderItems: OrderItem[];
  discounts?: Discount[];
}